src/Libraries/Microsoft.AspNetCore.Telemetry.Middleware/Logging/Internal/HttpLoggingRedactionInterceptor.cs

```diff
@@ -0,0 +1,206 @@
+        var request = context.Request;
+        if (ShouldExcludePath(context.Request.Path))
+        {
+            logContext.LoggingFields = HttpLoggingFields.None;
+        }
+        else if (logContext.IsAnyEnabled(HttpLoggingFields.Response))
+        {
+            // We'll need this for the response.
+            // TODO: Should we put a state filed on logContext?
+            context.Items[RequestStartTimestamp] = TimeProvider.GetTimestamp();
+        }
+
+        // Don't enrich if we're not going to log any part of the request
+        if (!logContext.IsAnyEnabled(HttpLoggingFields.Request))
+        {
+            return;
+        }
+
+        if (logContext.TryOverride(HttpLoggingFields.RequestPath))
+        {
+            string path = TelemetryConstants.Unknown;
+
+            if (_parameterRedactionMode != HttpRouteParameterRedactionMode.None)
+            {
+                var endpoint = context.GetEndpoint() as RouteEndpoint;
+
+                if (endpoint?.RoutePattern.RawText != null)
+                {
+                    var httpRoute = endpoint.RoutePattern.RawText;
+                    var paramsToRedact = _httpRouteUtility.GetSensitiveParameters(httpRoute, request, _parametersToRedactMap);
+
+                    var routeSegments = _httpRouteParser.ParseRoute(httpRoute);
+
+                    if (_requestPathLogMode == IncomingPathLoggingMode.Formatted)
+                    {
+                        path = _httpRouteFormatter.Format(in routeSegments, request.Path, _parameterRedactionMode, paramsToRedact);
+                    }
+                    else
+                    {
+                        // Case when logging mode is IncomingPathLoggingMode.Structured
+                        path = httpRoute;
+                        var routeParams = ArrayPool<HttpRouteParameter>.Shared.Rent(routeSegments.ParameterCount);
+
+                        // Setting this value right away to be able to return it back to pool in a callee's "finally" block:
+                        if (_httpRouteParser.TryExtractParameters(request.Path, in routeSegments, _parameterRedactionMode, paramsToRedact, ref routeParams))
+                        {
+                            foreach (var param in routeParams)
+                            {
+                                logContext.Add(param.Name, param.Value);
+                            }
+                        }
+                    }
+                }
+            }
+            else if (request.Path.HasValue)
+            {
+                path = request.Path.Value!;
+            }
+
+            logContext.Add(nameof(request.Path), path);
+        }
+
+        if (logContext.TryOverride(HttpLoggingFields.RequestHeaders))
+        {
+            // TODO: HttpLoggingOptions.Request/ResponseHeaders are ignored which could be confusing.
+            // Do we try to reconcile that with LoggingRedactionOptions.RequestHeadersDataClasses?
```

Comment:
I think this behavior as-is seems pretty reasonable and we'd document something like "Invoking AddHttpLoggingRedaction() offers more sophisticated redaction customization options for path and headers. The basic header configuration settings in HttpLoggingOptions will no longer be used when this is enabled."